theano/compile/debugmode.py

```diff
@@ -2781,6 +2828,9 @@ def __init__(self,
             # Copy to avoid sharing the same list across different instances
             self.check_preallocated_output = check_preallocated_output[:]
 
+        if check_overwrite_preallocated_data is not None:
+            self.check_overwrite_preallocated_data = check_overwrite_preallocated_data
```

Comment:
Add an assert, if check_overwrite_preallocated_data is True, then check_preallocated_output must be activated.